onnxruntime/core/optimizer/matmul_scale_fusion.cc

```diff
@@ -256,6 +256,7 @@ Status ProcessNode(
     }
   }
 
+  graph.Resolve();
   modified = true;
 
```

Comment:
GraphTransformer::Apply calls ApplyImpl, and does the graph.Resolve() if `modified` is true. Should only be necessary to call graph.Resolve() here if you need the graph to be resolved prior to processing more nodes/subgraphs in it.

Reply:
The problem we're seeing is that the test on L174 dereferences `node.Op()` on a "TransposeScaleMatMul" op that appears to have been added in an earlier call to this function, because `node.Op()` returns `nullptr`. 

Looking at the `GraphViewer` logic in `ApplyImpl()`, however, I'm not sure why a newly added node in the same pass would appear in the `node_indices`.

Reply:
You're right. The GraphViewer instance should have a fixed set of node indexes. There's no background sync with the underlying Graph class to add/remove values from that list. 

And AddNode should always use a new node index value, so a clash with an old/existing node index shouldn't be possible as we never attempt to reuse old index values.

Reply:
That makes sense! Anyway, @edgchen1 has an actual fix for the underlying problem in #4775, so this shouldn't be necessary. I'm going to drop this PR for now.